tests/loader/check_loader.py

```diff
@@ -96,7 +96,20 @@ def check_test_loader_with_class(self):
         tests = loader.discover([os.path.join(discover_dir(), "test_b.py")])
         assert len(tests) == 3
 
+    def check_test_loader_with_injected_args(self):
+        """When the --injected command-line option is used, the loader behaves a little bit differently:
 
+        each test method annotated with @parametrize or @matrix should only expand to a single discovered test,
+        and the injected args should be those passed in from command-line.
+        """
+        injected = {"x": 1, "y": -1}
+        loader = TestLoader(self.SESSION_CONTEXT, injected_args=injected)
 
+        file = os.path.join(discover_dir(), "test_decorated.py")
+        tests = loader.discover([file])
+        assert len(tests) == 4
```

Comment:
I found this behavior surprising and confusing. Should we really allow these injected parameters to apply to multiple tests? In this test, `test_decorated.py` happens to have tests that all take the same parameters, but the output wouldn't make much sense if the tests took different parameters (or none at all).

In fact, it won't even run properly if some tests don't have matching parameters unless test methods always include `**kwargs` in their argument list.

Should we restrict this option to only work when you specify a specific test method (not even just a single test class, but all the way to the specific method to ensure you're targeting the right test)?